### PR TITLE
feat: Allow Schema Serialization/deserialization

### PR DESCRIPTION
**Pull request by y0psolo**
This change allow Schema Serialzation/Deserialization based on the `into `and `try_from` parameters for serde from a new enum regrouping SchemaV1 and SchemaV2 struct. I added also some helper for conversion between the enum and Schema struct.

The functionality is tested with a new test inside the test section

### CODE REVIEW

**Comment by y0psolo** [comment]
close #41

File: crates/iceberg/src/spec/schema.rs
Comment (liurenjie1024) [inline crates/iceberg/src/spec/schema.rs:671]: Since the only difference between v1 and v2 for schema is `schema_id`, I think we should always convert to V2 since this is compatible with both formats. Using `schema_id == 0` may be incorrect since 0 maybe a valid schema  id .

Comment (y0psolo) [reply crates/iceberg/src/spec/schema.rs:671]: Thanks. I will update. Less code and much simpler i like it.
Moreover difference between V1 and V2 is very thin.

File: crates/iceberg/src/spec/schema.rs
Comment (liurenjie1024) [inline crates/iceberg/src/spec/schema.rs:749]: Could we also add some test for v1 without `schema-id`?

Comment (y0psolo) [reply crates/iceberg/src/spec/schema.rs]: Done, I implemented a test with schema-id set and with schema-id unset. I also added a check to be sure that V1 or V2 enum is correctly chosen depending if schema-id is set or unset.
I change the test schema to rely on the existing table_simple_schema already existing instead of relying on other one
I also tied the json string to table_simple_schema so that if we change the function, we also know we need to change the json string accordingly

**Review by Fokko — APPROVED** [review]
This makes sense to me. I'll let others review as well before merging 👍🏻

**Review by liurenjie1024 — APPROVED** [review]
LGTM, Thanks!

**Review by Fokko — APPROVED** [review]

**Pull request merged**
